lib/src/chart/base/axis_chart/axis_chart_data.dart

```diff
@@ -936,6 +942,21 @@ class HorizontalRangeAnnotation with EquatableMixin {
   /// If none is provided, it draws with a white color.
   final Gradient? gradient;
 
+  /// Text for annotation
+  final String? text;
+
+  /// Text style for annotation
+  final TextStyle style;
+
+  /// Text rotation in degrees
+  final double? rotation;
+
+  /// Text text padding
+  final double horizontalPadding;
```

Comment:
Also, padding is not what we have here. I think we can use `offset` name here. 
And I would define a single property `final Offset textOffset` for both horizontal and vertical variables (`dx` and `dy`)